### ISSUE TITLE
FormatParameterMismatch should ignore dynamic widths

### ISSUE THREAD

**Issue by deivid-rodriguez** [bbatsov/rubocop#2123]
```shell
$ cat bug.rb 
max_width = 7
id = 58
format('%*d', max_width, id)
```

```
$ rubocop bug.rb
Inspecting 1 file
W

Offenses:

bug.rb:3:1: W: Lint/FormatParameterMismatch: Number arguments (2) to format mismatches expected fields (1).
format('%*d', max_width, id)
^^^^^^

1 file inspected, 1 offense detected
```

This is a hard cop to get right...

**Comment by bbatsov** [comment]
//cc @edmz

**Comment by edmz** [comment]
I have been working on these, but they are a bit harder than what I originally thought.

**Comment by dreyks** [comment]
what about the opposite situation?
```ruby
v = [1, 2]
format('%d%d', *v)
```

` W: Number arguments (1) to format mismatches expected fields (2).`

this should probably be ignored too
